### ISSUE TITLE
Fix S2589 FP: Loop with Tuple Assignment

### ISSUE THREAD

**Issue by ThorstenReichert** [SonarSource/sonar-dotnet#3239]
### Description

When using a flag that is only updated through a tuple assignment, S2589 will be raised on conditions using that flag.

### Repro steps

```csharp
var tmp = 0;
var flag = true;
while(flag)
    (flag, tmp) = (false, 5);

Console.WriteLine(tmp);
```

### Expected behavior

No warning raised.

### Actual behavior

Raised warning: S2589: Change this condition so that it does not always evaluate to true.

### Related information

* SonarC# Version 8.5.0.15942
* Visual Studio Version: 16.4.6
* dotnet version: netcoreapp 3.1
* Operating System: Win10

**Comment by pavel-mikula-sonarsource** [comment]
This should be probably fixed by implementing #2933
